vendor/dataTables.datetime-moment.js

```diff
@@ -1,73 +1,51 @@
 /* eslint-disable no-undef */
-/**
- * This plug-in for DataTables represents the ultimate option in extensibility
- * for sorting date / time strings correctly. It uses
- * [Moment.js](http://momentjs.com) to create automatic type detection and
- * sorting plug-ins for DataTables based on a given format. This way, DataTables
- * will automatically detect your temporal information and sort it correctly.
- *
- * For usage instructions, please see the DataTables blog
- * post that [introduces it](//datatables.net/blog/2014-12-18).
- *
- * @name Ultimate Date / Time sorting
- * @summary Sort date and time in any format using Moment.js
- * @author [Allan Jardine](//datatables.net)
- * @depends DataTables 1.10+, Moment.js 1.7+
- *
- * @example
- *    $.fn.dataTable.moment( 'HH:mm MMM D, YY' );
- *    $.fn.dataTable.moment( 'dddd, MMMM Do, YYYY' );
- *
- *    $('#example').DataTable();
- */
 
+/* Updated by Fliplet on 2020-12-15 to work with ISO date format that is set in the data attribute 'isodate'
+ and do not have any bindings to the displayed date */
 (function(factory) {
   if (typeof define === 'function' && define.amd) {
     define(['jquery', 'moment', 'datatables.net'], factory);
   } else {
     factory(jQuery, moment);
   }
-}(function($, moment) {
-  $.fn.dataTable.moment = function( format, locale, reverseEmpties ) {
+})(function($, moment) {
+  $.fn.dataTable.moment = function(format) {
     var types = $.fn.dataTable.ext.type;
 
     // Add type detection
-    types.detect.unshift( function( d ) {
-      if ( d ) {
-        // Strip HTML tags and newline characters if possible
-        if ( d.replace ) {
-          d = d.replace(/(<.*?>)|(\r?\n|\r)/g, '');
-        }
-
-        // Strip out surrounding white space
-        d = $.trim( d );
+    types.detect.unshift(function(d) {
+      if (typeof d !== 'string') {
+        return null;
       }
 
-      if ( d === '' || d === null || d === '—' ) {
-        return 'moment-' + format;
+      // If we get match with sent format we check by momentjs that data attribute 'isodate' is a valid date
+      // If so we set type to our format
+      if (d.match(format)) {
+        return moment(
+          $(d)
```

Comment:
why is this repeated in the `span data-isodate` regexp?

Reply:
@squallstar Do you mean why we are using such a format?

Reply:
I don't understand the method we're using to fix this problem.

https://datatables.net/examples/advanced_init/html5-data-attributes.html suggests that by assigning a value for `data-order`, that value gets used for sorting. So it sounds like all we need to do is to add `:data-order="cellValue"` to the `<td>` then we're sorted.

Reply:
@tonytlwu For this way to work as you highlighted, we will need to remake how we are creating a table cell. Now we are putting content into the td element with vue v-for cycle, without knowing what data will be inside the td.
But it is possible to use your method with some changes. 
I'll remake the solution in a few moments and add a new commit.

Reply:
@tonytlwu Sorry for the long waiting. I've remade our solution and removed plugins at all. We don't them anymore. `data-order` attribute works just fine for our needs.
New [screencast](https://share.getcloudapp.com/KoulAzed)